model/task/task.go

```diff
@@ -1493,6 +1519,13 @@ func (t *Task) GetTaskLogs(ctx context.Context, env evergreen.Environment, getOp
 		return nil, errors.New("cannot get task logs for a display task")
 	}
 
+	output, ok := t.getTaskOutputSafe()
+	if !ok {
+		// We know there task cannot have task output, likely because
+		// it has not run yet. Return an empty iterator.
+		return log.EmptyIterator(), nil
```

Comment:
Did UI already confirm that in the case of no output, the UI will show the "No logs available" message even without an error?

Reply:
Yes! I looked at the UI code and confirmed with @khelif96 that this is the expected behavior.